calico/etcddriver/driver.py

```diff
@@ -170,7 +169,12 @@ def _handle_init(self, msg):
         # OK to dump the msg, it's a one-off.
         _log.info("Got init message from Felix %s", msg)
         self._etcd_base_url = msg[MSG_KEY_ETCD_URL].rstrip("/")
-        self._etcd_url_parts = urlparse(self._etcd_base_url)
+        self._etcd_scheme = msg[MSG_KEY_ETCD_URL].split(":")[0]
```

Comment:
hmm i see you pass in the scheme to urlparse ... so perhaps you can't seems odd though.